lib/site-inspector.rb

```diff
@@ -127,37 +134,213 @@ def to_json
     to_hash.to_json
   end
 
-  def to_hash
+  def http
+    details = {
+      # site-inspector's best guess
+      canonical: uri.to_s,
+      canonical_protocol: https? ? :https : :http,
+      canonical_endpoint: www? ? :www : :root,
+
+      endpoints: endpoints
+    }
+
+    combos = details[:endpoints]
+
+    details[:live] = (
+      combos[:https][:www][:live] or
+      combos[:https][:root][:live] or
+      combos[:http][:www][:live] or
+      combos[:http][:root][:live]
+    )
+
+    details[:broken_root] = (
+      (combos[:https][:root][:status] == 0) and
+      (combos[:http][:root][:status] == 0)
+    )
+
+    details[:broken_www] = (
+      (combos[:https][:www][:status] == 0) and
+      (combos[:http][:www][:status] == 0)
+    )
```

Comment:
A domain is broken on its `www` subdomain, or broken for its root, if _both_ the HTTP and HTTPS versions won't connect.